### PR TITLE
#171: Refactor app pages for static rendering pattern

### LINKED ISSUE #171: Refactor _app to not use getInitialProps
Should only need to get menu data using getStaticProps and allow app layer to be statically built. Move CTA to pages' data fetch.

### PR DESCRIPTION
**Pull request by rpeterman-gp**
Closes #171

- Updates all page types to be able to be rendered at build time (or as server-side render when not included in build paths.)
- The following pages are rendered at build time
  - Homepage
  - Stories links on homepage
  - Primary category links for stories on the homepage
  - Links in menus
  - Latest stories found in the sidebar of the homepage
  - Latest stories found on non-homepage pages' sidebar
  - The World program page
  - Episodes found on the first page of the episodes view of The World program page
  - Segment links for those episodes

## To Review

- [ ] Use the Preview link located in the Now comment below.

> ...then...

- [ ] Navigate around site and ensure pages load and render as expected. Should see faster load times.

### CODE REVIEW

**Comment by brandonhundt** [comment]
So fast! Passes my functional review. I wonder if we want to get someone else on the code review with a bit more React knowledge just to ensure that all looks good. (FYI @kookster)

**Review by brandonhundt — CHANGES_REQUESTED** [review]
Tag pages are returning a 500 error.

https://theworldorg-git-feat-171-refactor-app-no-getinitialpr-468d39.vercel.app/country/united-states

https://theworldorg-git-feat-171-refactor-app-no-getinitialpr-468d39.vercel.app/continent/north-america

**Comment by rpeterman-gp** [comment]
@brandonhundt Term pages should now load w/o error now.

**Review by brandonhundt — APPROVED** [review]
Passes functional review. Would like to see if we can get a more thorough code review before merging, if possible.

**Comment by brandonhundt** [comment]
@rpeterman-gp Great improvements thus far! When do you plan to call this PR complete?

**Comment by rpeterman-gp** [comment]
@brandonhundt I would like to call it done. Would like to have someone else familiar with webpack to have a look at at least the `next.config.js` before I merge this.

File: README.md [hunk 1]
```diff
@@ -200,13 +200,11 @@ First instinct would be to setup a catch-all route that can lookup content by al
...
+But what rewrites do we need? None of our content URL's adhere to any one pattern. There is one characteristic of a URL that makes things a little easier: they are _unique_ to the content they deliver from a data perspective. With a few exceptions for local app files and assets, we can assume all other URLs' pathnames are an alias for a page normally found on our Drupal CMS driven site. (This also assumes only requests for relevent front facing pages have been forward to this application. We shouldn't have to worry about admin pages, files, feeds, etc.) All we need is one rewrite to a [catch-all](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes) render page, `./pages/render/[...alias].tsx`.
```
Comment (svevang) [inline README.md:205]: This is really helpful!

File: README.md [hunk 2]
```diff
@@ -200,13 +200,11 @@ First instinct would be to setup a catch-all route that can lookup content by al
...
 
-The index page will be our whole app. Instead of routing to another page, it will be a proxy for our content components, dynamically loading and rendering the content component for the data associated with the alias. This will require mapping to get the right content component loaded.
-
-Mapping is handled in `./components/ResourceComponentMap.ts`. Content compnents are loaded using `next/dynamic` so pages don't have to import all pages components, just the one for that resource type. Simply add a dynamic import for your component, then add it to the `ResourceComponentMap` object, using the resource type as the key.
+The `render/[...alias].tsx` page will be our whole app, appart from the homepage, which remains our `index.tsx` page. Instead of routing to another page, it will be a proxy for our content components, dynamically loading and rendering the content component for the data type associated with the alias.
```
Comment (svevang) [inline README.md:207]: Really good to know, e.g for spelunking,  this is where one can start.

File: pages/render/[...alias].tsx [hunk 3]
```diff
@@ -3,72 +3,93 @@
...
+const DynamicAudio = dynamic(() => import('@components/pages/Audio'));
+const DynamicBio = dynamic(() => import('@components/pages/Bio'));
+const DynamicCategory = dynamic(() => import('@components/pages/Category'));
+const DynamicEpisode = dynamic(() => import('@components/pages/Episode'));
+const DynamicNewsletter = dynamic(() => import('@components/pages/Newsletter'));
+const DynamicPage = dynamic(() => import('@components/pages/Page'));
+const DynamicProgram = dynamic(() => import('@components/pages/Program'));
+const DynamicStory = dynamic(() => import('@components/pages/Story'));
+const DynamicTeam = dynamic(() => import('@components/pages/Team'));
+const DynamicTerm = dynamic(() => import('@components/pages/Term'));
```
Comment (svevang) [inline pages/render/[...alias].tsx:33]: Seeing the mapping of types here 👀

File: lib/fetch/category/fetchCategoryStories.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { fetchPriApiItem, fetchPriApiQuery } from '../api/fetchPriApi';
+import { basicStoryParams } from '../api/params';
+
+export const fetchCategoryStories = async (
+  id: string | IPriApiResource,
+  page: number = 1,
+  range: number = 15,
+  field = 'categories',
+  exclude: string[] | string = null
+): Promise<PriApiResourceResponse> => {
```
Comment (svevang) [inline lib/fetch/category/fetchCategoryStories.ts:21]: 👀 Part of the new api access pattern

File: store/actions/fetchPageData.ts [hunk 5]
```diff
@@ -30,7 +30,9 @@ export const fetchPageData = (
       }
     });
 
-    data = await fetchApiPage(id, req);
+    data = await fetchPage(id).then(
```
Comment (svevang) [inline store/actions/fetchPageData.ts:33]: 👀 the new fetching api

File: next.config.js [hunk 6]
```diff
@@ -48,6 +29,60 @@ module.exports = withPlugins([
           }
         }
       };
+
+      if (!isServer) {
```
Comment (svevang) [inline next.config.js:33]: Makes sense! The intent here is to split out overlapping frontend code into chunks -- eliminate dupe code!

Comment (svevang) [reply next.config.js:33]: Wild that this is `!isServer`, reminds me of meteor.js (back in the day!) ... code that runs on server and client (!!)

**Review by svevang — APPROVED** [review]
Thanks for the top level tour @rpeterman-gp. 😎 redux and react! 

The changes from getInitialProps to the new (eg. fetchPage) make sense (and 💨 are fast on production).  My react is rusty as all getout, but overall the pattern looks good and makes sense. QA and merge seems the way to go!

**Pull request merged**